forensics/Copy-VmDigitalEvidence.ps1

```diff
@@ -42,152 +42,190 @@ $ErrorActionPreference = 'Stop'
-
-######## Copy the OS disk hash value in key vault and delete disk in file share ##################
-Write-Output "#################################"
-Write-Output "OS disk - Put hash value in Key Vault"
-Write-Output "#################################"
-$secret = ConvertTo-SecureString -String $hash -AsPlainText -Force
-Set-AzKeyVaultSecret -VaultName $destKV -Name "$SnapshotName-sha256" -SecretValue $secret
-
-Remove-AzStorageFile -ShareName $destTempShare -Path $SnapshotName -Context $targetStorageContext
+    Get-AzSubscription -SubscriptionId $SubscriptionId | Set-AzContext
+    $vm = Get-AzVM -ResourceGroupName $ResourceGroupName -Name $VirtualMachineName
 
+    $disk = Get-AzDisk -ResourceGroupName $ResourceGroupName -DiskName $vm.StorageProfile.OsDisk.Name
+    $snapshot = New-AzSnapshotConfig -SourceUri $disk.Id -CreateOption Copy -Location $vm.Location
+    $snapshotName = $snapshotPrefix + "-" + $disk.name
+    New-AzSnapshot -ResourceGroupName $ResourceGroupName -Snapshot $snapshot -SnapshotName $snapshotname
 
-############################ Snapshot the data disks, store hash and BEK #####################
-$dsnapshotList = @()
 
-foreach ($dataDisk in $vm.StorageProfile.DataDisks) {
-    $ddisk = Get-AzDisk -ResourceGroupName $ResourceGroupName -DiskName $dataDisk.Name
-    $dsnapshot = New-AzSnapshotConfig -SourceUri $ddisk.Id -CreateOption Copy -Location $vm.Location
-    $dsnapshotName = $snapshotPrefix + "-" + $ddisk.name
-    $dsnapshotList += $dsnapshotName
-    Write-Output "Snapshot data disk name: $dsnapshotName"
-    New-AzSnapshot -ResourceGroupName $ResourceGroupName -Snapshot $dsnapshot -SnapshotName $dsnapshotName
-    
+    ##################### Copy the OS snapshot from source to file share and blob container ########################
     Write-Output "#################################"
-    Write-Output "Copy the Data Disk $dsnapshotName snapshot from source to blob container"
+    Write-Output "Copy the OS snapshot from source to file share and blob container"
     Write-Output "#################################"
 
-    $dsnapSasUrl = Grant-AzSnapshotAccess -ResourceGroupName $ResourceGroupName -SnapshotName $dsnapshotName -DurationInSecond 72000 -Access Read
-    $targetStorageContext = (Get-AzStorageAccount -ResourceGroupName $destRGName -Name $destSA).Context
+    $snapSasUrl = Grant-AzSnapshotAccess -ResourceGroupName $ResourceGroupName -SnapshotName $snapshotName -DurationInSecond 72000 -Access Read
+    Get-AzSubscription -SubscriptionId $destSubId | Set-AzContext
+    $targetStorageContextBlob = (Get-AzStorageAccount -ResourceGroupName $destRGName -Name $destSAblob).Context
+    $targetStorageContextFile = (Get-AzStorageAccount -ResourceGroupName $destRGName -Name $destSAfile).Context
+
+    Write-Output "Start Copying Blob $SnapshotName"
+    Start-AzStorageBlobCopy -AbsoluteUri $snapSasUrl.AccessSAS -DestContainer $destSAContainer -DestContext $targetStorageContextBlob -DestBlob "$SnapshotName.vhd" -Force

```

Comment:
Q: Why do we want to append `.vhd` here?  What problem does that solve? [Same Q for data disk below]

Reply:
When you need to mount the snapshot on a new virtual machine you must specify a .vhd file. By adding this extension the file is ready with the right extension and the user has less manual steps to do.

Reply:
Sounds good.  thanks!